tests/test_cache.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+import os
+
+from adr.configuration import Configuration
+
+
+def obtain_cache(config_file, contents):
+    config_file.write_text(contents)
+    return Configuration(path=config_file).cache
+
+
+def test_cache_with_file_store(tmp_path):
+    contents = (
+        """
+[adr]
+verbose = 1
+[adr.cache]
+default = "file"
+[adr.cache.stores]
+file = { driver = "file", path = "%s" }
+"""
+        % tmp_path
+    )
+    config_path = tmp_path / "config.toml"
+    cache = obtain_cache(config_path, contents)
+    cache.put("foo", "bar", 5)
+    assert cache.get("foo") == "bar"
+    os.remove(config_path)
+
+
+def test_cache_with_improper_config(tmp_path):
+    contents = """
+[adr]
+verbose = 1
+"""
+    config_path = tmp_path / "config.toml"
+    cache = obtain_cache(config_path, contents)
+    cache.put("foo", "bar", 5)
+    assert cache.get("foo") == "bar"
```

Comment:
This is the issue I've been facing all along. Unless the contents of config.toml are correct we can be using the Null store.

Should I change some code on Adr to prevent using the cache property unless the default store is different than the null store?

Reply:
Failing tests https://travis-ci.org/github/mozilla/mozci/builds/681058159